src/output-json.c

```diff
@@ -1268,3 +1370,26 @@ static void OutputJsonDeInitCtx(OutputCtx *output_ctx)
     SCFree(json_ctx);
     SCFree(output_ctx);
 }
+
+bool OutputJSONNeedFullLog(const OutputJsonCommonSettings *cfg, const Flow *f,
```

Comment:
`NeedsFullLog` seems to be able "stream data"?

Can you add a comment block indicating what the `type` param refers to and how this func is supposed to be used?

Reply:
Yes, currently streaming data is the main things added but it also triggers a full headers dump on HTTP for example.